sdk/python/feast/repo_config.py

```diff
@@ -128,6 +151,13 @@ def _validate_online_store_config(cls, values):
                 values["online_store"]["type"] = "sqlite"
             elif values["provider"] == "gcp":
                 values["online_store"]["type"] = "datastore"
+            elif values["provider"] == "redis":
+                values["online_store"]["type"] = "redis"
+                values["online_store"]["redis_connection_string"] = os.environ.get(
```

Comment:
@woop the connection string will contain password thus the ```feature_store.yaml``` is not the best place to keep secrets. In most cases the redis password should be kept in vault or in k8s secrets. The environment variable is the simplest way to share it with Feast, another solution is to set the value after ```FeatureStore``` instantiation (but I think it will be less intuitive for users).

Reply:
While I agree with that, I also think that we are adding complexity here that we can offload to the user for the time being. I think we'll need to have a separate PR by somebody in order to add env var handling to Feast. Right now, teams can still template their configuration file with the secret in production (which is actually safer than env vars since they are unlikely to be accidentally printed out) or they can fill in the secret themselves using a script during deployment.

What do you think? Can we delay this functionality for now?